File: daimler-front/src/app/services/passenger.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { UserRidesDTO, RideDTO, DepartureDestinationDTO, FavoriteRouteDTO } from '../entities/rides';
import { UserFull, UpdateUserDTO } from '../entities/user';

@Injectable({
  providedIn: 'root'
})
export class PassengerService {

  constructor(private http: HttpClient) { }

  get(id : number): Observable<UserFull> {
    return this.http.get<UserFull>(environment.apiHost + "passenger/" + id);
  }

  updatePassenger(id : Number, user : UpdateUserDTO) : Observable<any> {
    return this.http.put<UpdateUserDTO>(environment.apiHost + 'passenger/' + id, user);
  }

  getRides(id: number, page: number, size: number, sort: string, from: string, to: string) : Observable<UserRidesDTO>  {
    return this.http.get<UserRidesDTO>(environment.apiHost + "passenger/" + id + "/ride?page=" + page + "&size=" + size + "&sort=" + sort + "&from=" + from + "&to=" + to);
  }

  postFavoriteRoute(passengerId: number, departureId: number, destinationId: number) : Observable<FavoriteRouteDTO> {
    return this.http.post<FavoriteRouteDTO>(environment.apiHost + "ride/" + passengerId + "/favorite?departureId=" + departureId + "&destinationId=" + destinationId, {});
  }

  getFavoriteRoutes(passengerId: number) : Observable<FavoriteRouteDTO[]> {
    return this.http.get<FavoriteRouteDTO[]>(environment.apiHost + "ride/" + passengerId + "/favorite");
  }

  createRide(data: any) : Observable<RideDTO> {
    return this.http.post<RideDTO>(environment.apiHost + "ride", data);
  }

  getActiveRide(id: string) : Observable<any> {
    return this.http.get<any>(environment.apiHost + "ride/passenger/" + id + "/active");
  }

}

File: daimler-android/app/src/main/java/com/tim1/daimler/service/driver/PassengerService.java
package com.tim1.daimler.service.driver;

import com.tim1.daimler.dtos.passenger.FavoriteRouteDTO;
import com.tim1.daimler.dtos.ride.RidesDTO;
import com.tim1.daimler.dtos.user.UpdateUserDTO;
import com.tim1.daimler.dtos.user.UserDTO;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PassengerService {

    @GET("passenger/{id}")
    Call<UserDTO> getById(@Path("id") int id);

    @PUT("passenger/{id}")
    Call<UserDTO> update(@Path("id") int id, @Body UpdateUserDTO updated);

    @GET("ride/{id}/favorite")
    Call<List<FavoriteRouteDTO>> getFavoriteRoutes(@Path("id") int id);

    @DELETE("ride/{id}/favorite/{routeId}")
    Call<FavoriteRouteDTO> deleteFavoriteRoute(@Path("id") int id, @Path("routeId") int routeId);

    @GET("passenger/{id}/ride")
    Call<RidesDTO> getRides(@Path("id") int id, @Query("page") int page, @Query("size") int size, @Query("sort") String sort, @Query("from") String from, @Query("to") String to);

    @POST("ride/{id}/favorite")
    Call<FavoriteRouteDTO> addRouteToFavorite(@Path("id") int id, @Query("departureId") int departureId, @Query("destinationId") int destinationId);
}
